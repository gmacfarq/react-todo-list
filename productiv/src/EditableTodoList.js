import React from "react";
import EditableTodo from "./EditableTodo";

/** Show list of editable todos.
 *
 * Props:
 * - todos: array of [ todo, ... ]
 * - update(): fn to call to update a todo
 * - remove(): fn to call to remove a todo
 *
 * TodoApp -> EditableTodoList -> [ EditableTodo, ... ]
 */

function EditableTodoList({ todos, update, remove }) {
  console.log("todos list",todos)
  return (
    <div>
      {todos.map(t =>
        <EditableTodo
          key={t.id}
          todo={t}
          update={update}
          remove={() => remove(t.id)}
        />
      )}
    </div>
  );
}

export default EditableTodoList;
