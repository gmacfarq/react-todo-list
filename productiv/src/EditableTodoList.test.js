import { render, fireEvent } from "@testing-library/react";
import EditableTodoList from "./EditableTodoList";
import testTodos from "./initTodos";


const mock = () =>  { };


it("renders EditableTodo  without crashing", function () {
  render(
    <EditableTodoList
      todos={testTodos} update={mock()} remove={mock()}
/>
);
});

it("matches snapshot", function () {
const { container } = render(
  <EditableTodoList
    todos={testTodos} update={mock()} remove={mock()}
/>
);
expect(container).toMatchSnapshot();
});