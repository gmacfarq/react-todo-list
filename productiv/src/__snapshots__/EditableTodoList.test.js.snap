// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div>
    <div
      class="EditableTodo"
    >
      <div
        class="mb-3"
      >
        <div
          class="float-end text-sm-end"
        >
          <button
            class="EditableTodo-toggle btn-link btn btn-sm"
          >
            Edit
          </button>
          <button
            class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
          >
            Del
          </button>
        </div>
        <div
          class="Todo"
        >
          <div>
            <b>
              Code!
            </b>
             
            <small>
              2
            </small>
          </div>
          <div>
            <small>
              Write some code
            </small>
          </div>
        </div>
      </div>
    </div>
    <div
      class="EditableTodo"
    >
      <div
        class="mb-3"
      >
        <div
          class="float-end text-sm-end"
        >
          <button
            class="EditableTodo-toggle btn-link btn btn-sm"
          >
            Edit
          </button>
          <button
            class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
          >
            Del
          </button>
        </div>
        <div
          class="Todo"
        >
          <div>
            <b>
              Make dinner
            </b>
             
            <small>
              1
            </small>
          </div>
          <div>
            <small>
              Cook something healthy
            </small>
          </div>
        </div>
      </div>
    </div>
    <div
      class="EditableTodo"
    >
      <div
        class="mb-3"
      >
        <div
          class="float-end text-sm-end"
        >
          <button
            class="EditableTodo-toggle btn-link btn btn-sm"
          >
            Edit
          </button>
          <button
            class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
          >
            Del
          </button>
        </div>
        <div
          class="Todo"
        >
          <div>
            <b>
              Go to bed
            </b>
             
            <small>
              3
            </small>
          </div>
          <div>
            <small>
              In bed by 11:15
            </small>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
